includes/manage_newspaper.inc

```diff
@@ -19,7 +19,148 @@ require_once dirname(__FILE__) . '/utilities.inc';
+              '@pid' => $issue['pid'],
+            )
+          ),
+          '#attributes' => array('class' => array('datepicker')),
+          '#default_value' => (isset($form_state['input']['issue_dates'][$escaped_pid]) ? $form_state['input']['issue_dates'][$escaped_pid] : ""),
+          '#disabled' => (!islandora_user_access($issue_obj, array(ISLANDORA_METADATA_EDIT), array('islandora:newspaperIssueCModel'))),
+          '#size' => 20,
+          '#maxlength' => 10,
+        );
+      }
+    }
+    $form['submit'] = array(
+      '#type' => 'submit',
+      '#value' => t('Save'),
+    );
+    $form['issue_dates'] = array(
+      '#type' => 'value',
+    );
+    $form['#after_build'] = array(
+      'islandora_newspaper_empty_issue_date_datepicker',
+    );
+  }
+  else {
+    $form['dates_table']['all_good'] = array(
+      '#type' => 'item',
+      '#markup' => t('None found.'),
+    );
+  }
+  return $form;
+}
+
+/**
+ * Add datepicker to the issue date textfields.
+ *
+ * @param array $form
+ *   The form object.
+ * @param array $form_state
+ *   The form state
+ *
+ * @return array
+ *   Renderable array of the form.
+ */
+function islandora_newspaper_empty_issue_date_datepicker($form, $form_state) {
+  drupal_add_library('system', 'ui.datepicker');
+  drupal_add_js("(function ($) { $('.datepicker').datepicker({dateFormat: \"yy-mm-dd\"
+}); })(jQuery);", array('type' => 'inline', 'scope' => 'footer', 'weight' => 5));
+  return $form;
+}
+
+/**
+ * Implements hook_form_validate().
+ *
+ * Validate the dates entered on the form.
+ */
+function islandora_newspaper_empty_issue_date_form_validate($form, &$form_state) {
+  $errors = FALSE;
+  $updates = 0;
+  foreach ($form_state['values']['issue_dates'] as $key => $date) {
+    if (trim($date) != "") {
+      $updates += 1;
+      if (!preg_match('/^\d{4}-\d{1,2}-\d{1,2}$/', $date)) {
+        form_set_error("$key");
+        $errors = TRUE;
+      }
+      else {
+        try {
+          $dt = new DateTime($date);
+        }
+        catch (Exception $e) {
+          form_set_error("$key");
+          $errors = TRUE;
+        }
+      }
+    }
+  }
+  if ($errors) {
+    drupal_set_message(t("Invalid date(s), please enter a valid date of the format YYYY-MM-DD."), "error");
+  }
+  elseif ($updates == 0) {
+    drupal_set_message(t("You did not submit any dates."), "error");
+  }
+}
+
+/**
+ * Implements hook_form_submit().
+ *
+ * Update the objects with the dateIssued relationships.
+ */
+function islandora_newspaper_empty_issue_date_form_submit($form, &$form_state) {
+  module_load_include('inc', 'islandora_newspaper', 'includes/utilities');
+  foreach ($form_state['values']['issue_dates'] as $key => $date) {
+    if (trim($date) != "") {
+      $object = islandora_object_load($key);
+      if ($object) {
+        try {
+          $dt = new DateTime($date);
+          islandora_newspaper_set_date_issued($object, $dt);
+        }
+        catch (Exception $e) {
```

Comment:
Generally squashing all exceptions is inadvisable.  If there is a good reason for it then it should be documented to prevent someone from changing it.